File: app/src/main/java/cz/marvincz/rssnotifier/room/Converter.java
package cz.marvincz.rssnotifier.room;

import android.net.Uri;

import androidx.room.TypeConverter;

public class Converter {
    @TypeConverter
    public static String uriToString(Uri uri) {
        return uri.toString();
    }

    @TypeConverter
    public static Uri stringToUri(String string) {
        return Uri.parse(string);
    }
}

File: app/src/main/java/cz/marvincz/rssnotifier/room/Converter.kt
package cz.marvincz.rssnotifier.room

import android.net.Uri

import androidx.room.TypeConverter
import java.time.Instant
import java.time.ZoneId
import java.time.ZonedDateTime

class Converter {
    @TypeConverter
    fun uriToString(uri: Uri) = uri.toString()

    @TypeConverter
    fun stringToUri(string: String) = Uri.parse(string)!!

    @TypeConverter
    fun dateToLong(dateTime: ZonedDateTime): Long = dateTime.toInstant().toEpochMilli()

    @TypeConverter
    fun longToDate(long: Long): ZonedDateTime = ZonedDateTime.ofInstant(Instant.ofEpochMilli(long), ZoneId.systemDefault())
}
